core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -169,7 +169,15 @@ private void acquireConnectionAndExecute0(ClientRequestContext ctx, Endpoint end
                                               ClientConnectionTimingsBuilder timingsBuilder,
                                               ProxyConfig proxyConfig) {
         final PoolKey key = new PoolKey(endpoint, proxyConfig);
-        final HttpChannelPool pool = factory.pool(ctx.eventLoop().withoutContext());
+        final HttpChannelPool pool;
+        try {
+            pool = factory.pool(ctx.eventLoop().withoutContext());
+        } catch (Throwable t) {
+            final UnprocessedRequestException wrapped = UnprocessedRequestException.of(t);
+            handleEarlyRequestException(ctx, req, wrapped);
+            res.close(wrapped);
```

Comment:
I've noticed that similar logic is repeated in three different places within this class. Would you dedup them?

Reply:
I'll dedup similar logics 🏃